plugins/search-backend-module-explore/src/alpha.ts

```diff
@@ -23,66 +23,59 @@ import {
   coreServices,
   createBackendModule,
 } from '@backstage/backend-plugin-api';
-import { TaskScheduleDefinition } from '@backstage/backend-tasks';
 import { loggerToWinstonLogger } from '@backstage/backend-common';
 import { searchIndexRegistryExtensionPoint } from '@backstage/plugin-search-backend-node/alpha';
 
 import { ToolDocumentCollatorFactory } from '@backstage/plugin-search-backend-module-explore';
-
-/**
- * Options for {@link searchModuleExploreCollator}.
- *
- * @alpha
- */
-export type SearchModuleExploreCollatorOptions = {
-  schedule?: TaskScheduleDefinition;
-};
+import { readTaskScheduleDefinitionFromConfig } from '@backstage/backend-tasks';
 
 /**
  * Search backend module for the Explore index.
  *
  * @alpha
  */
-export const searchModuleExploreCollator = createBackendModule(
-  (options?: SearchModuleExploreCollatorOptions) => ({
-    moduleId: 'exploreCollator',
-    pluginId: 'search',
-    register(env) {
-      env.registerInit({
-        deps: {
-          config: coreServices.rootConfig,
-          logger: coreServices.logger,
-          discovery: coreServices.discovery,
-          scheduler: coreServices.scheduler,
-          tokenManager: coreServices.tokenManager,
-          indexRegistry: searchIndexRegistryExtensionPoint,
-        },
-        async init({
-          config,
-          logger,
-          discovery,
-          scheduler,
-          indexRegistry,
-          tokenManager,
-        }) {
-          const defaultSchedule = {
-            frequency: { minutes: 10 },
-            timeout: { minutes: 15 },
-            initialDelay: { seconds: 3 },
-          };
+export const searchModuleExploreCollator = createBackendModule({
+  moduleId: 'exploreCollator',
+  pluginId: 'search',
+  register(env) {
+    env.registerInit({
+      deps: {
+        config: coreServices.rootConfig,
+        logger: coreServices.logger,
+        discovery: coreServices.discovery,
+        scheduler: coreServices.scheduler,
+        tokenManager: coreServices.tokenManager,
+        indexRegistry: searchIndexRegistryExtensionPoint,
+      },
+      async init({
+        config,
+        logger,
+        discovery,
+        scheduler,
+        indexRegistry,
+        tokenManager,
+      }) {
+        const defaultSchedule = {
+          frequency: { minutes: 10 },
+          timeout: { minutes: 15 },
+          initialDelay: { seconds: 3 },
+        };
+
+        const schedule = config.has('search.explore.schedule')
```

Comment:
I've been moving these into the equivalent of `ToolDocumentCollatorFactory.fromConfig` instead, that way you can still use static configuration in the old system. Thinking that avoids potential confusion about why the config schema doesn't match and doesn't hurt much.